contracts/lpp/src/lpp.rs

```diff
@@ -796,8 +858,49 @@ mod test {
         assert_eq!(withdraw, Coin::new(1110));
     }
 
-    fn coin_cw(amount: u128) -> CwCoin {
-        coin_legacy::to_cosmwasm::<TheCurrency>(amount.into())
+    #[test]
+    fn test_deposit_limit() {
```

Comment:
a single positive test is not enough at all! I've spotted a bug only reading the code. You should have caught it with a test!

Reply:
With all due respect, I have a another test that does both positive and negative testing. It's in `lender.rs`.